builtin/docker/platform.go

```diff
@@ -141,43 +144,80 @@ func (p *Platform) Status(
 
 	log.Debug("querying docker for container health")
 
-	resources := []*sdk.StatusReport_Resource{{
-		Name: containerInfo.Name,
-	}}
+	// Find the declared resource that corresponds to our observed container
+	var containerDeclaredResource *sdk.DeclaredResource
+	for _, r := range declaredResources.Resources {
+		if r != nil && r.Name == "container" {
+			containerDeclaredResource = r
+		}
+	}
+
+	containerCreatedTime, err := time.Parse(time.RFC3339, containerInfo.Created)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to parse docker timestamp %q: %s", containerInfo.Created, err)
+	}
+
+	// Redact environment variables so we don't expose secrets on resources. Unmarshal/remarshaling to avoid mutating
+	// containerInfo
```

Comment:
If anyone knows of a less-clunky way to deep-copy this here i'm all ears.

Reply:
We usually use copystructure: https://github.com/mitchellh/copystructure